src/AutoRest.CSharp/Common/Input/Inputs.cs

```diff
@@ -120,38 +120,48 @@ internal record OperationPaging(string? NextLinkName, string? ItemName)
     }
 
-    internal record InputType(InputTypeKind Kind, bool IsNullable = false, InputTypeSerializationFormat SerializationFormat = InputTypeSerializationFormat.Default)
-    {
-        private readonly string? _name;
-        public InputType? ValuesType { get; init; }
-        public IReadOnlyList<InputTypeValue>? AllowedValues { get; init; }
-        public InputModel? Model { get; init; }
-
-        public string Name
-        {
-            get => _name ?? Model?.Name ?? Kind.ToString();
-            init => _name = value;
-        }
+    internal abstract record InputType(string Name, bool IsNullable = false) { }
 
-        public InputType() : this(InputTypeKind.Object) { }
-
-        public InputType(string name, InputTypeKind kind, bool isNullable, InputTypeSerializationFormat serializationFormat) : this(kind, isNullable, serializationFormat)
-        {
-            Name = name;
-        }
+    internal record InputPrimitiveType(InputTypeKind Kind, bool IsNullable = false) : InputType(Kind.ToString(), IsNullable)
+    {
+        public static InputPrimitiveType AzureLocation { get; }      = new(InputTypeKind.AzureLocation);
+        public static InputPrimitiveType Boolean { get; }            = new(InputTypeKind.Boolean);
+        public static InputPrimitiveType Bytes { get; }              = new(InputTypeKind.Bytes);
+        public static InputPrimitiveType BytesBase64Url { get; }     = new(InputTypeKind.BytesBase64Url);
+        public static InputPrimitiveType Date { get; }               = new(InputTypeKind.Date);
+        public static InputPrimitiveType DateTime { get; }           = new(InputTypeKind.DateTime);
+        public static InputPrimitiveType DateTimeISO8601 { get; }    = new(InputTypeKind.DateTimeISO8601);
+        public static InputPrimitiveType DateTimeRFC1123 { get; }    = new(InputTypeKind.DateTimeRFC1123);
+        public static InputPrimitiveType DateTimeUnix { get; }       = new(InputTypeKind.DateTimeUnix);
+        public static InputPrimitiveType DurationISO8601 { get; }    = new(InputTypeKind.DurationISO8601);
+        public static InputPrimitiveType DurationConstant { get; }   = new(InputTypeKind.DurationConstant);
+        public static InputPrimitiveType ETag { get; }               = new(InputTypeKind.ETag);
+        public static InputPrimitiveType Float32 { get; }            = new(InputTypeKind.Float32);
+        public static InputPrimitiveType Float64 { get; }            = new(InputTypeKind.Float64);
+        public static InputPrimitiveType Float128 { get; }           = new(InputTypeKind.Float128);
+        public static InputPrimitiveType Guid { get; }               = new(InputTypeKind.Guid);
+        public static InputPrimitiveType Int32 { get; }              = new(InputTypeKind.Int32);
+        public static InputPrimitiveType Int64 { get; }              = new(InputTypeKind.Int64);
+        public static InputPrimitiveType Object { get; }             = new(InputTypeKind.Object);
```

Comment:
For a primitive type, what does `InputPrimitiveType.Object` mean? A `Map<K, V>`?

Reply:
'unknown'. I don't know if we ever see this in real CADL, but since it is defined as one of the possible types, we should have something that would match it. `Map<K, V>` is retired in favor of `Record<T>`.